hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/audit/impl/LoggingAuditor.java

```diff
@@ -230,6 +240,25 @@ private class LoggingAuditSpan extends AbstractAuditSpanImpl {
 
     private final HttpReferrerAuditHeader referrer;
 
+    /**
+     * Attach Range of data for GetObject Request
+     * @param request given get object request
+     */
+    private void attachRangeFromRequest(AmazonWebServiceRequest request) {
+      if (request instanceof GetObjectRequest) {
+        long[] rangeValue = ((GetObjectRequest) request).getRange();
+        if (rangeValue == null || rangeValue.length == 0) {
+          return;
+        }
+        if (rangeValue.length != 2) {
+          WARN_INCORRECT_RANGE.warn("Expected range to contain 0 or 2 elements. Got "
+              + rangeValue.length + ". Ignoring");
+        }
+        String combinedRangeValue = String.format("bytes=%d-%d", rangeValue[0], rangeValue[1]);
```

Comment:
These 2 lines should only be executed when `length == 2`.

Maybe you're missing a `return` in the `if` branch above?